src/tests/class_tests/openms/source/IdentificationData_test.cpp

```diff
@@ -0,0 +1,386 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/SYSTEM/SysInfo.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+
+#include <OpenMS/METADATA/ID/IdentificationData.h>
+
+///////////////////////////
+
+START_TEST(IdentificationData, "$Id$")
+
+/////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////
+
+using namespace OpenMS;
+using namespace std;
+
+IdentificationData* ptr = 0;
+IdentificationData* null = 0;
+START_SECTION((IdentificationData()))
+  ptr = new IdentificationData();
+  TEST_NOT_EQUAL(ptr, null);
+END_SECTION
+
+IdentificationData data;
+IdentificationData::InputFileRef file_ref;
+IdentificationData::ProcessingSoftwareRef sw_ref;
+IdentificationData::SearchParamRef param_ref;
+IdentificationData::ProcessingStepRef step_ref;
+IdentificationData::ScoreTypeRef score_ref;
+IdentificationData::DataQueryRef query_ref;
+IdentificationData::ParentMoleculeRef protein_ref, rna_ref;
+IdentificationData::IdentifiedPeptideRef peptide_ref;
+IdentificationData::IdentifiedOligoRef oligo_ref;
+IdentificationData::IdentifiedCompoundRef compound_ref;
+IdentificationData::QueryMatchRef match_ref1, match_ref2, match_ref3;
+
+START_SECTION((const InputFiles& getInputFiles() const))
+{
+  TEST_EQUAL(data.getInputFiles().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((InputFileRef registerInputFile(const String& file)))
+{
+  String file = "test.mzML";
+  file_ref = data.registerInputFile(file);
+  TEST_EQUAL(data.getInputFiles().size(), 1);
+  TEST_STRING_EQUAL(*file_ref, file);
+}
+END_SECTION
+
+START_SECTION((const DataProcessingSoftware& getDataProcessingSoftware() const))
+{
+  TEST_EQUAL(data.getDataProcessingSoftware().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((ProcessingSoftwareRef registerDataProcessingSoftware(const Software& software)))
+{
+  Software sw("Tool", "1.0");
+  sw_ref = data.registerDataProcessingSoftware(sw);
+  TEST_EQUAL(data.getDataProcessingSoftware().size(), 1);
+  TEST_EQUAL(*sw_ref == sw, true); // "TEST_EQUAL(*sw_ref, sw)" doesn't compile
```

Comment:
can you also test if you register the same tool twice, you should get the same reference back (e.g. no double entries, right) ?

Reply:
I've now added that check wherever it's relevant.